src/config.rs

```diff
@@ -139,21 +146,65 @@ impl ConfigDir {
         config_dir
     }
 
+    pub fn extend(&self, branch: &str) -> Result<ConfigDir, Error> {
+        match self {
+            ConfigDir::Git {
+                url, ssh_key_path, ..
+            } => ConfigDir::new(
+                ConfigUrl::Git {
+                    url: url.clone(),
+                    branch: Some(branch.to_owned()),
+                    internal_path: PathBuf::new(),
+                },
+                ssh_key_path,
+            ),
+            ConfigDir::File { .. } => Err(format_err!("Can not extend file config")),
+        }
+    }
+
     pub fn directory(&self) -> &Path {
         match *self {
             ConfigDir::File { ref directory, .. } => directory,
             ConfigDir::Git { ref directory, .. } => directory,
         }
     }
 
-    // TODO: Implement being able to re-checkout a git repo
-    pub fn refresh(&self) -> &Self {
-        match *self {
-            ConfigDir::File { .. } => {}
-            ConfigDir::Git { .. } => {}
+    pub fn refresh(&self, target: Option<&str>) -> Option<String> {
+        match self {
+            ConfigDir::File { .. } => None,
+            ConfigDir::Git {
+                directory,
+                url,
+                ssh_key_path,
+                ..
+            } => {
+                let git_repo = match git::build_repo(directory.to_str().unwrap()) {
+                    Ok(repo) => repo,
+                    Err(e) => {
+                        error!(
+                            "Error: {} \n Unable to find the git repo: {}",
+                            e,
+                            directory.to_str().unwrap()
+                        );
+                        return None;
+                    }
+                };
+                match git::reset(
+                    &git_repo,
+                    "origin/master",
```

Comment:
This should probably be the default branch rather than hardcoded to master

Reply:
We don't leave it on master, all routes require a sha to be specified. I can see if the lib will let me derive the default branch.

Reply:
Any update here?